.github/workflows/test.yml

```diff
@@ -1,31 +1,92 @@
-name: Test
+name: Tests
 
 # Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the master branch
+  # Triggers the workflow on push or pull request events
   push:
-    branches: [ tests ]
   pull_request:
-    branches: [ tests ]
 
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-    
-  # This workflow contains a single job called "tests"
-  tests:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
+  windows:
+    strategy:
+      fail-fast: false
+      matrix:
+        python: [python38, python39]
+        rfVersion: [3.1.2, 3.2, 3.2.1, 3.2.2]
+    runs-on: windows-latest
+    name: Windows (${{ matrix.python }}, robotframework-${{ matrix.rfVersion }})
+    defaults:
+      run:
+        shell: wsl-bash {0}
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - uses: actions/checkout@v2
+      - uses: Vampire/setup-wsl@v1
+        with:
+          distribution: Ubuntu-20.04
```

Comment:
I would appreciate a comment here, because right now I had to take a double-take like "wait... are we running on Windows or Ubuntu wtf"

So an explaining comment along the lines of "Running on Windows inside Windows Subsystem for Linux (WSL)"